tests/spyre_util.py

```diff
@@ -383,6 +387,24 @@ def create_random_request(request_id: int,
         if seed is not None:
             random.seed(seed)
         prompt_token_ids = random.choices(valid_token_ids, k=num_tokens)
+
+        if prefix_len is not None and prefix_len < num_tokens:
+            token_id_to_replace = prompt_token_ids[prefix_len]
+            # resample a token to diverge the prompt after prefix_len
+            while True:
+                token_id_replacement = random.choices(valid_token_ids, k=1)[0]
+                if token_id_replacement != token_id_to_replace:
+                    prompt_token_ids[prefix_len] = token_id_replacement
+                    break
+        if n_lookahead is not None and n_lookahead > 0:
```

Comment:
I think we need an assert here that `prefix_len` is also not None and > 0 to avoid error in line 407

Reply:
oh you checked both of those above, good job